### ISSUE TITLE
Provide Docker container

### ISSUE THREAD

**Issue by abergmeier** [github/hub#2340]
For e.g. using Hub with CloudBuild, it would be useful to have a prepackaged container.
Created one for myself here: https://hub.docker.com/r/abergmeier/hub

Would be better though, if this was an official container.

**Comment by mislav** [comment]
Sorry, you will have to explain your use-case better.

I see you've built `hub` into an `alpine/git` container here https://github.com/abergmeier/hub/commit/5db89d53c3f5d99ce9581a7ff849cdfe40e4e854

And now? How do you invoke `hub` from within this container?

**Comment by abergmeier** [comment]
> How do you invoke hub from within this container?

E.g. by `docker run abergmeier/hub:2.12.8 release ...`

Services like CloudBuild is entirely running on Docker Containers. Thus this would be helpful.

**Comment by mislav** [comment]
> E.g. by `docker run abergmeier/hub:2.12.8 release ...`

Ok. And if I wanted to attach a file with `docker run abergmeier/hub:2.12.8 release create -a myfile.tgz` to a release, how would hub running within a container get access to `myfile.tgz` that is on the filesystem of the host process?

Also, plenty of other hub commands need to access the current git repository. How would hub running within `docker run abergmeier/hub:2.12.8 pull-request <args>` be able to access the git repository in the current working directory of the process that spawned `docker run`? The filesystems inside and outside of the container are isolated.

I just don't get how I can package hub in a container, let people use that official container, and still allow people to use `hub` with all the same functionality as when running outside of a container. Maybe you can explain that to me.

**Comment by abergmeier** [comment]
Sorry, I sometimes forget that now everybody is knee-deep in container world.

> how would hub running within a container get access to myfile.tgz that is on the filesystem of the host process

Well there are multiple scenarios. Basically for most container runtimes, you have a way of mounting volumes into the container.

For docker you e.g. would say `-v /myhostdir:/mycurrentcontainerdir` and then have access to a potential `/myhostdir/myfile.tgz` inside the container (as`/mycurrentcontainerdir/myfile.tgz`) .

For CloudBuild you e.g. would save `myfile.tgz` into a CloudStorage and then copy the files onto a shared volume in a previous build step.

> The filesystems inside and outside of the container are isolated.

They are not totally isolated. And as shown above, you can deliberately introduce filesystem data into your container.

So basically what you do is prepare an environment that has enough data, so that git/hub can work.

> Maybe you can explain that to me.

Hope that makes it clearer.

**Comment by mislav** [comment]
Thank you for explaining!

**Comment by AndreasBergmeier6176** [comment]
Just FYI - added support for OpenShift Containers, too.

**Comment by antmak** [comment]
Hi! Where is an current version `hub` on Docker Hub? Official version? I could't find

**Comment by mislav** [comment]
There is not an official container image for hub and there likely will never be one because hub as a tool is designed to have access to your host system and will not be happy when run from inside any container. See this rationale given for GitHub CLI: https://github.com/cli/cli/discussions/3698#discussioncomment-785838 (it also applies to hub because they are very similar tools).
